FALL_DETECTION.ipynb

```diff
@@ -0,0 +1,587 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0e2c3330-e095-46e2-b7a3-986156cb619e",
+   "metadata": {},
+   "source": [
+    "### Data Collection and Preparation\n",
+    "####  Data Sources\n",
+    "The dataset used in this project consists of time-series data collected from wearable devices, specifically accelerometer and gyroscope sensors. The sensors operate at a frequency of 50Hz, capturing detailed motion data. The data contains multiple fall scenarios along with routine movements, which helps in training the model to recognize fall patterns.\n",
+    "#### Data Merging and Cleaning\n",
+    "To combine the accelerometer and gyroscope data, the datasets were merged on the timestamp for each activity. Gyroscope time lists were discarded to avoid redundancy, and missing values were carefully handled by removing NaNs, ensuring data integrity for the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 54,
+   "id": "fac9088b-591a-42f5-8318-42227dea371d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "All files merged and labeled successfully.\n",
+      "Final merged dataset saved with user and fall type information.\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import pandas as pd\n",
+    "\n",
+    "# Function to merge accel and gyro data\n",
+    "def merge_data(accel_file, gyro_file):\n",
+    "    accel_data = pd.read_csv(accel_file)\n",
+    "    gyro_data = pd.read_csv(gyro_file)\n",
+    "    merged_data = pd.concat([accel_data, gyro_data.drop(columns=['gyro_time_list'])], axis=1)  # drop gyro_time_list\n",
+    "    return merged_data\n",
+    "\n",
+    "# Function to label the merged data based on timestamps\n",
+    "def apply_labels(merged_data, filename, fall_timestamps):\n",
+    "    # Filter rows for the current filename\n",
+    "    file_timestamps = fall_timestamps[fall_timestamps['filename'] == filename]\n",
+    "\n",
+    "    # Initialize the labels column with 0\n",
+    "    merged_data['labels'] = 0\n",
+    "\n",
+    "    # Loop through each timestamp entry for this file\n",
+    "    for _, row in file_timestamps.iterrows():\n",
+    "        start_time = row['start_time']\n",
+    "        end_time = row['end_time']\n",
+    "\n",
+    "        # Assign label 1 if accel_time_list falls within the start_time and end_time\n",
+    "        merged_data.loc[(merged_data['accel_time_list'] >= start_time) & (merged_data['accel_time_list'] <= end_time), 'labels'] = 1\n",
+    "\n",
+    "    return merged_data\n",
+    "\n",
+    "# Load the fall timestamps data\n",
+    "fall_timestamps_path = '/Users/rhythmnarula/Desktop/WEDA-FALL-main/dataset/fall_timestamps.csv'\n",
```

Comment:
Can you please remove the filepath to your personal desktop? Instead add a dummy path and make note in README file to give instructions on how to setup the folders.